zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/UsageMetricsQueryControllerTest.java

```diff
@@ -0,0 +1,109 @@
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+
+@WebMvcTest(UsageMetricsQueryController.class)
+public class UsageMetricsQueryControllerTest extends RestControllerTest {
+  static final String USAGE_METRICS_URL = "/v2/usage-metrics/";
+  static final String USAGE_METRICS_SEARCH_URL = USAGE_METRICS_URL + "search";
+
+  static final String EXPECTED_SEARCH_RESPONSE =
+      """
+      {
+         "assignees": 5,
+         "processInstances": 23,
+         "decisionInstances": 17
+      }""";
+
+  static final UsageMetricsCount USAGE_METRICS_COUNT_ENTITY = new UsageMetricsCount(5L, 23L, 17L);
+
+  @MockBean UsageMetricsServices usageMetricsServices;
+
+  @BeforeEach
+  void setupUsageMetricsServices() {
+    when(usageMetricsServices.withAuthentication(any(Authentication.class)))
+        .thenReturn(usageMetricsServices);
+  }
+
+  @Test
+  void shouldSearchWithStartTimeAndEndTime() {
+    // given
+    when(usageMetricsServices.search(any(UsageMetricsQuery.class)))
+        .thenReturn(USAGE_METRICS_COUNT_ENTITY);
+    final var request =
+        """
+        {
+           "startTime": "1970-11-14T10:50:26.000Z",
+           "endTime":   "2024-12-31T10:50:26.000Z"
+        }
+        """;
+    webClient
+        .post()
+        .uri(USAGE_METRICS_SEARCH_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_JSON)
+        .expectBody()
+        .json(EXPECTED_SEARCH_RESPONSE);
+
+    final var startTime = OffsetDateTime.of(1970, 11, 14, 10, 50, 26, 0, ZoneOffset.UTC);
+    final var endTime = OffsetDateTime.of(2024, 12, 31, 10, 50, 26, 0, ZoneOffset.UTC);
+
+    verify(usageMetricsServices)
+        .search(
+            new UsageMetricsQuery.Builder()
+                .filter(
+                    new UsageMetricsFilter.Builder()
+                        .startTime(Operation.gte(startTime))
+                        .endTime(Operation.lte(endTime))
+                        .build())
+                .build());
+  }
+
+  @Test
+  void shouldThrowExceptionIfNoStartAndEndTimeAreGiven() {
+    webClient
+        .post()
+        .uri(USAGE_METRICS_SEARCH_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue("{}")
+        .exchange()
+        .expectStatus()
+        .isBadRequest()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_PROBLEM_JSON_VALUE)
+        .expectBody()
+        .json(
+            "{\"type\":\"about:blank\",\"title\":\"INVALID_ARGUMENT\",\"status\":400,\"detail\":\"startTime and endTime must be specified.\",\"instance\":\"/v2/usage-metrics/search\"}");
```

Comment:
🔧 This is quite hard to read, how about we use multiline strings like the other test cases?

Reply:
Adjusting this to use multiline strings